File: net46/Misc/BarMap.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="BarMap.cs" company="Inspire IT Ltd">
//   Copyright © Inspire IT Ltd. All Rights Reserved.
// </copyright>
// --------------------------------------------------------------------------------------------------------------------

namespace RyanPenfold.TestProject.Repository.FluentMappings
{
    using FluentNHibernate.Mapping;

    using Model;

    /// <summary>
    /// Defines a mapping for type <see cref="Bar"/>.
    /// </summary>
    public class BarMap : ClassMap<Bar>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="BarMap"/> class.
        /// </summary>
        public BarMap()
        {
            this.Id(x => x.Id).GeneratedBy.Assigned();
            this.Map(x => x.BigIntColumn);
            this.Map(x => x.BinaryColumn);
            this.Map(x => x.BitColumn);
            this.Map(x => x.CharColumn);
            this.Map(x => x.DateColumn).CustomType("date");
            this.Map(x => x.DateTimeColumn);
            this.Map(x => x.DateTime2Column).CustomType("datetime2");
            this.Map(x => x.DateTimeOffsetColumn);
            this.Map(x => x.DecimalColumn);
            this.Map(x => x.FloatColumn);
            this.Map(x => x.GeographyColumn);
            this.Map(x => x.GeometryColumn);
            this.Map(x => x.HierarchyIdColumn);
            this.Map(x => x.ImageColumn);
            this.Map(x => x.IntColumn);
            this.Map(x => x.MoneyColumn);
            this.Map(x => x.NCharColumn);
            this.Map(x => x.NTextColumn);
            this.Map(x => x.NumericColumn);
            this.Map(x => x.NVarCharColumn);
            this.Map(x => x.RealColumn);
            this.Map(x => x.SmallDateTimeColumn);
            this.Map(x => x.SmallIntColumn);
            this.Map(x => x.SmallMoneyColumn);
            this.Map(x => x.SqlVariantColumn);
            this.Map(x => x.TextColumn);
            this.Map(x => x.TimeColumn).CustomType("TimeAsTimeSpan");
            this.Map(x => x.TimestampColumn).ReadOnly();
            this.Map(x => x.TinyIntColumn);
            this.Map(x => x.UniqueIdentifierColumn);
            this.Map(x => x.VarBinaryColumn);
            this.Map(x => x.VarCharColumn);
            this.Map(x => x.XmlColumn);
            this.Table("Bar");
            this.Schema("dbo");
        }
    }
}

File: net45/Misc/BarMap.vb
' --------------------------------------------------------------------------------------------------------------------
' <copyright file="BarMap.vb" company="Inspire IT Ltd">
'   Copyright © Inspire IT Ltd. All Rights Reserved.
' </copyright>
' --------------------------------------------------------------------------------------------------------------------

Imports FluentNHibernate.Mapping

Imports RyanPenfold.TestProject.Model

Namespace FluentMappings

''' <summary>
''' Defines a mapping for type <see cref="Bar"/>.
''' </summary>
Public Class BarMap
	Inherits ClassMap(Of Bar)

	''' <summary>
	''' Initializes a new instance of the <see cref="BarMap"/> class.
	''' </summary>
	Public Sub New()
		Id(Function(x) x.Id).GeneratedBy.Assigned()
		Map(Function(x) x.BigIntColumn)
		Map(Function(x) x.BinaryColumn)
		Map(Function(x) x.BitColumn)
		Map(Function(x) x.CharColumn)
		Map(Function(x) x.DateColumn).CustomType("date")
		Map(Function(x) x.DateTimeColumn)
		Map(Function(x) x.DateTime2Column).CustomType("datetime2")
		Map(Function(x) x.DateTimeOffsetColumn)
		Map(Function(x) x.DecimalColumn)
		Map(Function(x) x.FloatColumn)
		Map(Function(x) x.GeographyColumn)
		Map(Function(x) x.GeometryColumn)
		Map(Function(x) x.HierarchyIdColumn)
		Map(Function(x) x.ImageColumn)
		Map(Function(x) x.IntColumn)
		Map(Function(x) x.MoneyColumn)
		Map(Function(x) x.NCharColumn)
		Map(Function(x) x.NTextColumn)
		Map(Function(x) x.NumericColumn)
		Map(Function(x) x.NVarCharColumn)
		Map(Function(x) x.RealColumn)
		Map(Function(x) x.SmallDateTimeColumn)
		Map(Function(x) x.SmallIntColumn)
		Map(Function(x) x.SmallMoneyColumn)
		Map(Function(x) x.SqlVariantColumn)
		Map(Function(x) x.TextColumn)
		Map(Function(x) x.TimeColumn).CustomType("TimeAsTimeSpan")
		Map(Function(x) x.TimestampColumn).ReadOnly()
		Map(Function(x) x.TinyIntColumn)
		Map(Function(x) x.UniqueIdentifierColumn)
		Map(Function(x) x.VarBinaryColumn)
		Map(Function(x) x.VarCharColumn)
		Map(Function(x) x.XmlColumn)
		Table("Bar")
		Schema("dbo")
	End Sub

End Class

End Namespace